src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -91,6 +92,21 @@ public class SocketOutput : ISocketOutput
                 if (buffer.Count > 0)
                 {
                     var tail = ProducingStart();
+                    if (tail.IsDefault)
+                    {
+                        if (cancellationToken.CanBeCanceled)
+                        {
+                            return TaskUtilities.GetCancelledTask(
+                                cancellationToken.IsCancellationRequested ?
+                                cancellationToken :
+                                new CancellationToken(true));
```

Comment:
Its sort of the approach from `Listening to Multiple Tokens Simultaneously` from https://msdn.microsoft.com/en-us/library/dd997364(v=vs.110).aspx with the internal and external token but without the expense of using `CreateLinkedTokenSource`; happy to change it pending discussion with @Tratcher